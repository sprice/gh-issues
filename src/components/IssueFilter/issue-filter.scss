.issue-filter {
  max-width: $breakpoint-x-large;
  margin: 2rem auto 2rem;
  display: flex;
  justify-content: space-between;
}

.filters {
  margin-left: 1.5rem;
  display: flex;
  flex-direction: column;
}

.filters a {
  border: 0;
  background-color: $white;
  margin-right: 0.5rem;
  color: $dark-grey;
  text-decoration: none;
}

.filters a:hover {
  color: $grey;
}

.filters a.active {
  color: $pink;
}

i.close::before {
  content: "";
  display: block;
  width: 20px;
  height: 20px;
  margin-right: 20px;

  background-color: $dark-grey;
  mask-image: url("../icons/close.svg");
  mask-repeat: no-repeat;
  -webkit-mask-image: url("../icons/close.svg");
  -webkit-mask-repeat: no-repeat;
}

@media (min-width: $breakpoint-small) {
  .filters {
    flex-direction: row;
  }
}
