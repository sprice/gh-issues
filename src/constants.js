export default {
  FILTER: {
    ALL: "all",
    OPEN: "open",
    CLOSED: "closed",
    PULL: "pull_request"
  },
  GH_API_BASE: "https://api.github.com/"
}
