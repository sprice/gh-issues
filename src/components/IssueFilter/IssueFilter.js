import React from "react"
import PropTypes from "prop-types"
import { Link } from "@reach/router"
import classNames from "classnames"
import constants from "../../constants"

class IssueFilter extends React.Component {
  handleClickAll = () => {
    this.props.getIssues(
      this.props.owner,
      this.props.repo,
      constants.FILTER.ALL
    )
  }

  handleClickOpen = () => {
    this.props.getIssues(
      this.props.owner,
      this.props.repo,
      constants.FILTER.OPEN
    )
  }

  handleClickClosed = () => {
    this.props.getIssues(
      this.props.owner,
      this.props.repo,
      constants.FILTER.CLOSED
    )
  }

  handleClickPull = () => {
    this.props.getIssues(
      this.props.owner,
      this.props.repo,
      constants.FILTER.PULL
    )
  }

  render() {
    const allIsActive = this.props.visibilityFilter === constants.FILTER.ALL
    const all = classNames({ active: allIsActive })

    const openIsActive = this.props.visibilityFilter === constants.FILTER.OPEN
    const open = classNames({ active: openIsActive })

    const closedIsActive =
      this.props.visibilityFilter === constants.FILTER.CLOSED
    const closed = classNames({ active: closedIsActive })

    const pullIsActive = this.props.visibilityFilter === constants.FILTER.PULL
    const pull = classNames({ active: pullIsActive })

    const baseUrl = `/viewer/${this.props.owner}/${this.props.repo}/`

    return (
      <div className="issue-filter">
        <div className="filters">
          <Link
            to={`${baseUrl}${constants.FILTER.ALL}`}
            className={all}
            onClick={this.handleClickAll}
          >
            All Issues
          </Link>
          <Link
            to={`${baseUrl}${constants.FILTER.OPEN}`}
            className={open}
            onClick={this.handleClickOpen}
          >
            Open Issues
          </Link>
          <Link
            to={`${baseUrl}${constants.FILTER.CLOSED}`}
            className={closed}
            onClick={this.handleClickClosed}
          >
            Closed Issues
          </Link>
          <Link
            to={`${baseUrl}${constants.FILTER.PULL}`}
            className={pull}
            onClick={this.handleClickPull}
          >
            Pull Requests
          </Link>
        </div>
        <div className="close-viewer">
          <Link to="/" onClick={this.props.clearState}>
            <i className="icon close" />
          </Link>
        </div>
      </div>
    )
  }
}

IssueFilter.propTypes = {
  getIssues: PropTypes.func.isRequired,
  owner: PropTypes.string.isRequired,
  repo: PropTypes.string.isRequired,
  visibilityFilter: PropTypes.string.isRequired,
  clearState: PropTypes.func.isRequired
}

export default IssueFilter
