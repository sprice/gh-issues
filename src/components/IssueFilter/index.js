import IssueFilter from "./IssueFilter"

export default IssueFilter
