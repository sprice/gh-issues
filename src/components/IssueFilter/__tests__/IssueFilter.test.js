import React from "react"
import { create } from "react-test-renderer"
import IssueFilter from "../IssueFilter"
import constants from "../../../constants"

test("Snapshop of IssueFilter", () => {
  const c = create(
    <IssueFilter
      owner="ethereum"
      repo="solidity"
      visibilityFilter={constants.FILTER.OPEN}
    />
  )
  expect(c.toJSON()).toMatchSnapshot()
})
