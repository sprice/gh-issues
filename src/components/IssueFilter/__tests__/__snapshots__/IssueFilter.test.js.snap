// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshop of IssueFilter 1`] = `
<div
  className="issue-filter"
>
  <div
    className="filters"
  >
    <a
      className=""
      href="/viewer/ethereum/solidity/all"
      onClick={[Function]}
    >
      All Issues
    </a>
    <a
      className="active"
      href="/viewer/ethereum/solidity/open"
      onClick={[Function]}
    >
      Open Issues
    </a>
    <a
      className=""
      href="/viewer/ethereum/solidity/closed"
      onClick={[Function]}
    >
      Closed Issues
    </a>
    <a
      className=""
      href="/viewer/ethereum/solidity/pull_request"
      onClick={[Function]}
    >
      Pull Requests
    </a>
  </div>
  <div
    className="close-viewer"
  >
    <a
      aria-current="page"
      href="/"
      onClick={[Function]}
    >
      <i
        className="icon close"
      />
    </a>
  </div>
</div>
`;
